third_party/gfxreconstruct/framework/decode/vulkan_dive_consumer_base.h

```diff
@@ -0,0 +1,163 @@
+                                                   DescriptorUpdateTemplateDecoder* pData) override
+    {
+        Process_vkUpdateDescriptorSetWithTemplate(
+            call_info, device, descriptorSet, descriptorUpdateTemplate, pData, false);
+    }
+
+  protected:
+    void Process_vkUpdateDescriptorSetWithTemplate(const ApiCallInfo&               call_info,
+                                                   format::HandleId                 device,
+                                                   format::HandleId                 descriptorSet,
+                                                   format::HandleId                 descriptorUpdateTemplate,
+                                                   DescriptorUpdateTemplateDecoder* pData,
+                                                   bool                             use_KHR_suffix);
+
+    virtual void Process_vkCmdPushDescriptorSetWithTemplateKHR(const ApiCallInfo& call_info,
+                                                               format::HandleId   commandBuffer,
+                                                               format::HandleId   descriptorUpdateTemplate,
+                                                               format::HandleId   layout,
+                                                               uint32_t           set,
+                                                               DescriptorUpdateTemplateDecoder* pData) override;
+
+    virtual void Process_vkCmdPushDescriptorSetWithTemplate2KHR(
+        const ApiCallInfo&                                                 call_info,
+        format::HandleId                                                   commandBuffer,
+        StructPointerDecoder<Decoded_VkPushDescriptorSetWithTemplateInfo>* pPushDescriptorSetWithTemplateInfo) override;
+
+    void WriteBlockStart() { writer_->WriteBlockStart(); }
+                                      
+    void WriteBlockEnd(util::DiveFunctionData function_data) { writer_->WriteBlockEnd(function_data); }
+
+
+    /// A field not present in binary format which identifies the index of each
+    /// command within its command buffer.
+    /// @todo Make this field optional.
+    constexpr const char* NameCommandIndex() const { return "cmd_index"; }
+    /// A field not present in binary format which identifies the index of each
+    /// submit in the global order of all submits to all queues as recorded in
+    /// the binary trace file.
+    /// @todo Make this field optional.
+
+    uint32_t GetCommandBufferRecordIndex(format::HandleId command_buffer)
+    {
+        uint32_t index = ++rec_cmd_index_[command_buffer];
+        return index;
+    }
+
+    void ResetCommandBufferRecordIndex(format::HandleId command_buffer) { rec_cmd_index_[command_buffer] = 0; }
+
+    uint32_t                                       submit_index_{ 0 }; // index of submissions across the trace
```

Comment:
Any reason that those members are public?

Reply:
Removed the submit_index_ member since it is no longer needed and made the other members private.